### PR TITLE
Run rustfmt and enforce via CI

### PR DESCRIPTION
**Pull request by fintelia**
Now seems like a good time to run rustfmt over the codebase. Happy to add more #[rustfmt::skip] annotations

### CODE REVIEW

File: src/buffer.rs [hunk 1]
```diff
@@ -1516,88 +1533,163 @@ mod benchmarks {
...
+        assert!(!image.copy_within(
+            Rect {
+                x: 1,
+                y: 1,
+                width: 4,
+                height: 4
+            },
+            0,
+            0
+        ));
```
Comment (HeroicKatora) [inline src/buffer.rs:1605]: Similar concerns of readability / formatting of buffers below.

File: src/codecs/jpeg/decoder.rs [hunk 2]
```diff
@@ -195,28 +197,1060 @@ mod tests {
...
+            [126, 63, 189],
+            [122, 61, 183],
+            [118, 59, 177],
+            [114, 57, 171],
+            [106, 53, 159],
+            [98, 49, 147],
+            [94, 47, 141],
+            [86, 43, 129],
+            [150, 100, 50],
+            [98, 49, 245],
```
Comment (HeroicKatora) [inline src/codecs/jpeg/decoder.rs:1252]: Mkay, not sure if I like it or not.

File: src/codecs/jpeg/encoder.rs [hunk 3]
```diff
@@ -946,13 +922,23 @@ mod tests {
...
                 1, // density is in dpi
-                300u16.to_be_bytes()[0], 300u16.to_be_bytes()[1],
-                300u16.to_be_bytes()[0], 300u16.to_be_bytes()[1],
-                0, 0, // No thumbnail
+                300u16.to_be_bytes()[0],
+                300u16.to_be_bytes()[1],
+                300u16.to_be_bytes()[0],
+                300u16.to_be_bytes()[1],
+                0,
+                0, // No thumbnail
```
Comment (HeroicKatora) [inline src/codecs/jpeg/encoder.rs:941]: Grumble, but still fine.

File: src/codecs/webp/vp8.rs [hunk 4]
```diff
@@ -2660,19 +2707,26 @@ mod test {
...
+                        floor_avg,
+                        avg3(i, j, k)
+                    );
                 }
             }
         }
     }
 
     #[test]
     fn test_edge_pixels() {
```
Comment (HeroicKatora) [inline src/codecs/webp/vp8.rs:2726]: Needs some skip for these, I think, to preserve the structure of prediction inputs.

**Review by HeroicKatora — COMMENTED** [review]
A few spots that should be skipped but you're right, now or not for a long time.

**Comment by fintelia** [comment]
Seems that the CI script isn't installing rustfmt properly... :confused: 
```
info: checking for self-updates
warning: tool `rustfmt` is already installed, remove it from `/home/runner/.cargo/bin`, then run `rustup update` to have rustup manage this tool.
warning: tool `cargo-fmt` is already installed, remove it from `/home/runner/.cargo/bin`, then run `rustup update` to have rustup manage this tool.
```

**Comment by fintelia** [comment]
...which actually turns out not to be the issue. The CI [already has rustfmt version 1.4.37 installed](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#packages). Rather it was just a merge conflict with another PR

**Review by HeroicKatora — APPROVED** [review]

**Pull request merged**